From b74d8e1883858f8b6c50411dfeeceafbb2018e79 Mon Sep 17 00:00:00 2001
From: "Brad T. Peters" <brad.t.peters@intel.com>
Date: Tue, 28 Jun 2016 22:25:03 +0000
Subject: [PATCH] Make isZeroU handle the V256 case. Fixes #356393

Patch from:
github.com/liquid-mirror/VEX/commit/9602ceb38e8f73c124943f289ba2effda4b80d54

git-svn-id: svn://svn.valgrind.org/vex/trunk@3213 8f6e269a-dfd6-0310-a8e1-e2731360e62c

Signed-off-by: Brad T. Peters <brad.t.peters@intel.com>
---
 VEX/priv/ir_opt.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/VEX/priv/ir_opt.c b/VEX/priv/ir_opt.c
index 7a7246a..4266823 100644
--- a/VEX/priv/ir_opt.c
+++ b/VEX/priv/ir_opt.c
@@ -1223,6 +1223,7 @@ static Bool isZeroU ( IRExpr* e )
       case Ico_U16:   return toBool( e->Iex.Const.con->Ico.U16 == 0);
       case Ico_U32:   return toBool( e->Iex.Const.con->Ico.U32 == 0);
       case Ico_U64:   return toBool( e->Iex.Const.con->Ico.U64 == 0);
+      case Ico_V256:  return toBool( e->Iex.Const.con->Ico.V256 == 0x00000000);
       default: vpanic("isZeroU");
    }
 }
-- 
2.8.1

